test/mlu_op_gtest/pb_gtest/include/dispatch_enumDtype.h

```diff
@@ -0,0 +1,174 @@
+ * Copyright (C) [2025] by Cambricon, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *************************************************************************/
+#pragma once
+#include <variant>
+
+#include "mlu_op.h"
+
+namespace Eigen {
+struct half;
+struct bfloat16;
+}  // namespace Eigen
+
+using Eigen::bfloat16;
+using Eigen::half;
+
+template <typename T>
+struct TypeToEnum {
+  static constexpr mluOpDataType_t value = MLUOP_DTYPE_INVALID;
+};
+
+#define typeMapEnum(type, enum)                    \
+  template <>                                      \
+  struct TypeToEnum<type> {                        \
+    static constexpr mluOpDataType_t value = enum; \
+  }
+
+typeMapEnum(float, MLUOP_DTYPE_FLOAT);
+typeMapEnum(half, MLUOP_DTYPE_HALF);
+typeMapEnum(bfloat16, MLUOP_DTYPE_BFLOAT16);
+typeMapEnum(double, MLUOP_DTYPE_DOUBLE);
+typeMapEnum(int8_t, MLUOP_DTYPE_INT8);
+typeMapEnum(int16_t, MLUOP_DTYPE_INT16);
+typeMapEnum(int32_t, MLUOP_DTYPE_INT32);
+typeMapEnum(int64_t, MLUOP_DTYPE_INT64);
+typeMapEnum(uint8_t, MLUOP_DTYPE_UINT8);
+typeMapEnum(uint16_t, MLUOP_DTYPE_UINT16);
+typeMapEnum(uint32_t, MLUOP_DTYPE_UINT32);
+typeMapEnum(uint64_t, MLUOP_DTYPE_UINT64);
+typeMapEnum(bool, MLUOP_DTYPE_BOOL);
+typeMapEnum(std::complex<half>, MLUOP_DTYPE_COMPLEX_HALF);
+typeMapEnum(std::complex<float>, MLUOP_DTYPE_COMPLEX_FLOAT);
+
+/*
+ * Parameter Explanation: T1 must be provided as the default type,
+ *  and ...Types represents the other supported types.
+ *
+ * Key Technology: Variadic templates, using fold expressions.
+ *
+ * Notes: A default type must be provided as the initial variable;
+ *  otherwise, if multiple types do not have a common initial value to assign,
+ * an error will occur. If an empty type is used as a placeholder, there will
+ * also be an empty type in the function dispatch, which could result in an
+ * error due to the missing corresponding instantiation.
+ */
```

Comment:
换行通常缩进俩空

```suggestion
 * Parameter Explanation: T1 must be provided as the default type,
 *   and ...Types represents the other supported types.
 *
 * Key Technology: Variadic templates, using fold expressions.
 *
 * Notes: A default type must be provided as the initial variable;
 *   otherwise, if multiple types do not have a common initial value to assign,
 *   an error will occur. If an empty type is used as a placeholder, there will
 *   also be an empty type in the function dispatch, which could result in an
 *   error due to the missing corresponding instantiation.
 */
```